sql/core/src/main/scala/org/apache/spark/sql/execution/Columnar.scala

```diff
@@ -454,6 +454,7 @@ case class RowToColumnarExec(child: SparkPlan) extends UnaryExecNode {
 
           override def next(): ColumnarBatch = {
             cb.setNumRows(0)
+            vectors.foreach(_.reset())
```

Comment:
This bug seems to exist in `branch-2.4`, doesn't it? With a proper UT, we need to backport this to `branch-2.4`. Please update the JIRA issue `Affects Versions` together.

Reply:
Thanks for your comments. Sorry but don't find this bug in `branch-2.4`. This part of code is proposed by [https://issues.apache.org/jira/browse/SPARK-27396](url)

Reply:
You are right, @rongma1997 . It seems that I misread the commit history last time.